src/Xamarin.Android.Build.Tasks/Utilities/XAJavaTypeScanner.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+//using System.IO.MemoryMappedFiles;
+
+using Java.Interop.Tools.Cecil;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks;
+
+class JavaType
+{
+	public readonly TypeDefinition Type;
+	public readonly IDictionary<AndroidTargetArch, TypeDefinition>? PerAbiTypes;
+	public bool IsABiSpecific { get; }
+
+	public JavaType (TypeDefinition type, IDictionary<AndroidTargetArch, TypeDefinition>? perAbiTypes)
+	{
+		Type = type;
+		if (perAbiTypes != null) {
+			PerAbiTypes = new ReadOnlyDictionary<AndroidTargetArch, TypeDefinition> (perAbiTypes);
```

Comment:
Why create a new `ReadOnlyDictionary` wrapper?

Reply:
Defensive coding - I don't want that data set changed by accident or otherwise

Reply:
That will prevent people from modifying `javaType.PerAbiTypes`, but that won't defend against people modifying the underlying collection in `perAbiTypes`.  This `new ReadOnlyDictionary` doesn't "deep copy" `perAbiTypes`, it *shares the reference*.  Other code could thus modify `perAbiTypes` (if they have an appropriate reference) to add/remove items.

Reply:
I need "simple" protection against adding/removing entries to this dictionary. `ReadOnlyDictionary` gives that, it's good enough.